Json.More/TypeResolverOptionsManager.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System.Text.Encodings.Web;
+using System.Text.Json;
+using System.Text.Json.Serialization.Metadata;
+
+namespace Json.More;
+
+public class TypeResolverOptionsManager
+{
+	private JsonSerializerOptions? _serializerOptions;
+	private JsonSerializerOptions? _serializerOptionsUnsafeRelaxedJsonEscaping;
+#if NET8_0_OR_GREATER
+	private readonly IJsonTypeInfoResolver _baseResolver;
+	private IJsonTypeInfoResolver _typeInfoResolver;
+#endif
+	private readonly object _serializerOptionsLock = new();
+	
+	public JsonSerializerOptions SerializerOptions
+	{
+		get
+		{
+			lock (_serializerOptionsLock)
+			{
+				_serializerOptions ??= new JsonSerializerOptions
+				{
+#if NET8_0_OR_GREATER
+					TypeInfoResolver = _typeInfoResolver
+#endif
+				};
+
+				return _serializerOptions!;
+			}
+		}
+	}
+
+	public JsonSerializerOptions SerializerOptionsUnsafeRelaxedJsonEscaping
+	{
+		get
+		{
+			lock (_serializerOptionsLock)
+			{
+				_serializerOptionsUnsafeRelaxedJsonEscaping ??= new JsonSerializerOptions
+				{
+					Encoder = JavaScriptEncoder.UnsafeRelaxedJsonEscaping,
+#if NET8_0_OR_GREATER
+					TypeInfoResolver = _typeInfoResolver
+#endif
+				};
+
+				return _serializerOptionsUnsafeRelaxedJsonEscaping!;
+			}
+		}
+	}
+
+#if NET8_0_OR_GREATER
+	public IJsonTypeInfoResolver TypeInfoResolver => _typeInfoResolver;
+	
+	public TypeResolverOptionsManager(IJsonTypeInfoResolver baseResolver, params IJsonTypeInfoResolver[] resolvers)
+	{
+		_baseResolver = baseResolver;
+		_typeInfoResolver = JsonTypeInfoResolver.Combine([baseResolver, .. resolvers]);
+	}
+
+	public void RebuildTypeResolver(params IJsonTypeInfoResolver[] resolvers)
```

Comment:
The reason I had this as "invalidate" was so that we didn't do extra work on each Register call and only built the list the next time someone asked for it. Since the behavior is maintained for the options objects, it's probably ok.

Reply:
But the code was also holding onto references outside of itself.